File: Civitas-Java/Civitas/src/main/java/civitas/Dado.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package civitas;

import java.util.Random;

/**
 * @author elena
 * @date 26/09/2019
 */

public class Dado {    
    private static final Dado instance = new Dado();
    private static int SalidaCarcel = 5;
    private int random;
    private int ultimoResultado;
    private boolean debug;
      
    public static Dado getInstance(){
        return instance;
    }
  
    private Dado(){
        SalidaCarcel = 5;
        ultimoResultado = 0;
        random = 0;
        debug = false;
    }
    
    int tirar(){ 
        Random rand = new Random();
        if(!debug){
            ultimoResultado = rand.nextInt(4);
        }
        else
            ultimoResultado = 1;
        
        return ultimoResultado;
    }
    
    boolean salgoDeLaCarcel(){
        int dado = tirar();
        
        return dado >= 5;
    }
    
    int quienEmpieza(int n){
        Random rand = new Random();
        return rand.nextInt(n);
    }
    
    public void setDebug(boolean d){
        Diario diariodado = Diario.getInstance();
        debug = d;
        
        if(d)   diariodado.ocurreEvento("Debug activo");
        else    diariodado.ocurreEvento("Debug desactivado");
    }
    
    int getUltimoResultado(){
        return ultimoResultado;
    }
}

File: Civitas-Ruby/lib/dado.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.
require 'singleton'
require_relative "diario"

module Civitas
  class Dado
    include Singleton

    def initialize
      @random = 0
      @ultimoResultado = 0
      @debug = false
      @@SalidaCarcel = 5
    end
    
    public
    def tirar
      if !@debug
        @ultimoResultado = rand(4)
      else
        @ultimoResultado = 1
      end
      
      @ultimoResultado
    end
    
    def salgodelacarcel
      tirar >= 5
    end
    
    def quienempieza(n)
      rand(n)
    end
    
    def setdebug(d)
      @debug = d
      diariodado = Diario.instance
      if d
        diariodado.ocurre_evento("Debug activo")
      else
        diariodado.ocurre_evento("Debug desactivado")
      end
    end
    
    attr_reader :ultimoResultado
  end
end
